frontend-react/src/__mocks__/DBMock.test.ts

```diff
@@ -0,0 +1,239 @@
+                    undefined,
+                ],
+            },
+            {
+                label: "relativePathMap post",
+                args: [
+                    "https://localhost",
+                    { post: "/settings/organizations" },
+                    undefined,
+                ],
+            },
+            {
+                label: "parentResolver getList",
+                args: [
+                    "https://localhost",
+                    undefined,
+                    { getList: parentResolver },
+                ],
+            },
+            {
+                label: "parentResolver get by primary key",
+                args: ["https://localhost", undefined, { get: parentResolver }],
+            },
+            {
+                label: "parentResolver post",
+                args: [
+                    "https://localhost",
+                    undefined,
+                    { post: parentResolver },
+                ],
+            },
+            {
+                label: "relativePathMap and parentResolver getList",
+                args: [
+                    "https://localhost",
+                    { getList: "/settings/organizations" },
+                    { getList: parentResolver },
+                ],
+            },
+            {
+                label: "relativePathMap and parentResolver by primary key",
+                args: [
+                    "https://localhost",
+                    { get: "/settings/organizations/:name" },
+                    { get: parentResolver },
+                ],
+            },
+            {
+                label: "relativePathMap and parentResolver post",
+                args: [
+                    "https://localhost",
+                    { post: "/settings/organizations" },
+                    { post: parentResolver },
+                ],
+            },
+        ];
+
+        // In case a test errors, cleanup before next one
+        afterEach(() => {
+            mockServer?.close();
+            mockServer &&= undefined;
+        });
+
+        test.each(restHandlerScenarios)("$label", async ({ args }) => {
+            const [basePath, relativePathMap, parentResolverMap] = args;
+            db.organizationSettings.createMany(10);
+
+            const handlers = db.organizationSettings.toEnhancedRestHandlers(
+                ...args
+            );
+            const handlerMap = factoryRestHandlerTupleToMap(handlers);
+            mockServer = setupServer(...handlers);
+            mockServer.listen();
+
+            const records = db.organizationSettings.getAllJson();
+            const methods = Object.keys(relativePathMap ?? parentResolverMap ?? {}) as RestHandlerMethods[];
+            // If no custom paths/resolvers given, we're just testing a default handler
+            if (methods.length === 0) {
+                methods.push("getList");
+            }
+
+            for (const method of methods) {
```

Comment:
This is another place where breaking up into dedicated tests rather than using `test.each` could help with overall clarity.  I don't think we should have a lot of `if-else`s (or any, really) when we're running expectations -- each logical fork should be its own `describe` or `test` in my opinion.